auth0/src/test/java/com/auth0/android/authentication/storage/CredentialsManagerTest.java

```diff
@@ -344,6 +345,106 @@ public void shouldGetNonExpiredCredentialsFromStorageWhenOnlyAccessTokenIsAvaila
         assertThat(retrievedCredentials.getScope(), is("scope"));
     }
 
+    @Test
+    public void shouldRenewCredentialsWhenScopeHasChanged() {
+        when(storage.retrieveString("com.auth0.id_token")).thenReturn("idToken");
+        when(storage.retrieveString("com.auth0.access_token")).thenReturn("accessToken");
+        when(storage.retrieveString("com.auth0.refresh_token")).thenReturn("refreshToken");
+        when(storage.retrieveString("com.auth0.token_type")).thenReturn("type");
+        long expirationTime = CredentialsMock.CURRENT_TIME_MS + 123456L * 1000; //non expired credentials
+        when(storage.retrieveLong("com.auth0.expires_at")).thenReturn(expirationTime);
+        when(storage.retrieveLong("com.auth0.cache_expires_at")).thenReturn(expirationTime);
+        when(storage.retrieveString("com.auth0.scope")).thenReturn("some new scope");
+        when(client.renewAuth("refreshToken")).thenReturn(request);
+
+        Date newDate = new Date(CredentialsMock.CURRENT_TIME_MS + 1234 * 1000);
```

Comment:
Why not use the same `expirationTime`?

Reply:
I could but, in theory, shouldn't the new tokens have a newer date? I'll refactor these into constants and use something like NOW and TOMORROW

Reply:
I've ended up using CURRENT_TIME_MS and added one new that adds one hour on top of that ONE_HOUR_AHEAD_MS. Looks cleaner now.